src/crnlib.rs

```diff
@@ -0,0 +1,144 @@
+
+    // DXT5 alpha blocks only
+    CCrnfmtDxt5a,
+
+    CCrnfmtEtc1,
+    CCrnfmtEtc2,
+    CCrnfmtEtc2a,
+    CCrnfmtEtc1s,
+    CCrnfmtEtc2as,
+
+    CCrnfmtTotal,
+
+    CCrnfmtForceDword = 0xFFFFFFFF
+}
+
+#[repr(C)]
+pub struct CrnTextureInfo {
+    pub m_struct_size: u32,
+    pub m_width: u32,
+    pub m_height: u32,
+    pub m_levels: u32,
+    pub m_faces: u32,
+    pub m_bytes_per_block: u32,
+    pub m_userdata0: u32,
+    pub m_userdata1: u32,
+    pub m_format: CrnFormat
+}
+
+impl CrnTextureInfo{
+    pub fn default() -> CrnTextureInfo{
+        return CrnTextureInfo {
+            m_struct_size: core::mem::size_of::<CrnTextureInfo>() as u32,
+			m_width: 0,
+			m_height: 0,
+			m_levels: 0,
+			m_faces: 0,
+			m_bytes_per_block: 0,
+			m_userdata0: 0,
+			m_userdata1: 0,
+			m_format: CrnFormat::CCrnfmtInvalid // Init as invalid?
+        }
+    }
+
+    pub fn crnd_get_texture_info(&mut self, p_data: &[u8], data_size: u32) -> bool{
+        if data_size < core::mem::size_of::<CrnHeader>() as u32 {
+            return false;
+        }
+
+        if self.m_struct_size != core::mem::size_of::<CrnTextureInfo>() as u32{
+            return false;
+        }
+
+        let mut p_header: CrnHeader = CrnHeader::default();
+        let res: bool = p_header.crnd_get_header(p_data, data_size); 
+        if res == false {
+            return res;
+        }
+
+        self.m_width = p_header.m_width.cast_to_uint();
+        self.m_height = p_header.m_height.cast_to_uint();
+        self.m_levels = p_header.m_levels.cast_to_uint();
+        self.m_faces = p_header.m_faces.cast_to_uint();
+        self.m_format = match p_header.m_format.cast_to_uint(){
+            // -1 => crn_format::cCRNFmtInvalid,
+
+            0 => CrnFormat::CCrnfmtDxt1,
+        
+            // 0 => crn_formatcCRNFmtFirstValid,
+        
+            // cCRNFmtDXT3 is not currently supported when writing to CRN - only DDS.
+            1 => CrnFormat::CCrnfmtDxt3,
+        
+            2 => CrnFormat::CCrnfmtDxt5,
+        
+            // Various DXT5 derivatives
+            3 => CrnFormat::CCrnfmtDxt5CcxY,    // Luma-chroma
+            4 => CrnFormat::CCrnfmtDxt5XGxR,    // Swizzled 2-component
+            5 => CrnFormat::CCrnfmtDxt5XGbr,    // Swizzled 3-component
+            6 => CrnFormat::CCrnfmtDxt5Agbr,    // Swizzled 4-component
+        
+            // ATI 3DC and X360 DXN
+            7 => CrnFormat::CCrnfmtDxnXy,
+            8 => CrnFormat::CCrnfmtDxnYx,
+        
+            // DXT5 alpha blocks only
+            9 => CrnFormat::CCrnfmtDxt5a,
+        
+            10 => CrnFormat::CCrnfmtEtc1,
+            11 => CrnFormat::CCrnfmtEtc2,
+            12 => CrnFormat::CCrnfmtEtc2a,
+            13 => CrnFormat::CCrnfmtEtc1s,
+            14 => CrnFormat::CCrnfmtEtc2as,
+        
+            15 => CrnFormat::CCrnfmtTotal,
+        
+            0xFFFFFFFF => CrnFormat::CCrnfmtForceDword,
+
+            _ => CrnFormat::CCrnfmtInvalid

```

Comment:
replace with `_ => return false,` and remove the if after this match

```suggestion
            _ => return false,
```